server/backend/testspg/crud/test_user_validator.py

```diff
@@ -0,0 +1,138 @@
+    "user_guid": USER_GUID_BCEID_LOAD_2_TEST,
+}
+
+
+class TestUserValidatorClass(object):
+    """
+    Testing UserValidator class with real remote API calls (TEST environment).
+    """
+
+    def setup_class(self):
+        # local valid mock requester
+        self.requester_idir = Requester(**TEST_USER_IDIR)
+
+    def test_verify_user_exist_idir(self):
+        target_user = TargetUser(**TEST_USER_IDIR)
+        user_validaor = UserValidator(self.requester_idir, target_user)
+        verified_target_user = user_validaor.verify_user_exist()
+        # test the verified target user
+        assert verified_target_user.user_guid == target_user.user_guid
+        assert verified_target_user.user_type_code == target_user.user_type_code
+        assert verified_target_user.user_name == target_user.user_name
+
+    def test_verify_user_exist_idir_not_found(self):
+        target_user = TargetUser(**{**TEST_USER_IDIR, "user_name": "USER_NOT_EXISTS"})
+        user_validaor = UserValidator(self.requester_idir, target_user)
+
+        with pytest.raises(HTTPException) as e:
+            user_validaor.verify_user_exist()
+        assert (
+            str(e.value.detail.get("code")).find(ERROR_CODE_INVALID_REQUEST_PARAMETER)
+            != -1
+        )
+        assert (
+            str(e.value.detail.get("description")).find(
+                "Invalid request, cannot find user"
+            )
+            != -1
+        )
+
+    def test_verify_user_exist_idir_mismatch_info(self):
+        target_user = TargetUser(
+            **{
+                **TEST_USER_IDIR,
+                "user_guid": "USERGUIDNOTEXISTSPOJHSLEJFNSEKSL",
+            }
+        )
+        user_validaor = UserValidator(self.requester_idir, target_user)
+
+        with pytest.raises(HTTPException) as e:
+            user_validaor.verify_user_exist()
+        assert (
+            str(e.value.detail.get("code")).find(ERROR_CODE_INVALID_REQUEST_PARAMETER)
+            != -1
+        )
+        assert (
+            str(e.value.detail.get("description")).find("does not match the user guid")
+            != -1
+        )
+
+    def test_verify_user_exist_bceid(self):
+        target_user = TargetUser(**TEST_TARGET_USER_BCEID_LOAD_2)
+        user_validaor = UserValidator(self.requester_idir, target_user)
+        verified_target_user = user_validaor.verify_user_exist()
+        # test the verified target user, business guid is added
+        assert verified_target_user.user_guid == target_user.user_guid
+        assert verified_target_user.user_type_code == target_user.user_type_code
+        assert verified_target_user.user_name == target_user.user_name
+        assert verified_target_user.business_guid is not None
+
+    # def test_verify_user_exist_bceid_not_found(self):
+    #     target_user = TargetUser(
+    #         **{
+    #             **TEST_TARGET_USER_BCEID_LOAD_2,
+    #             "user_guid": "USERGUIDNOTEXISTSPOJHSLEJFNSEKSL",
```

Comment:
Skip the two tests for now, cause we suppose to search bceid on user_guid, but because that's not enabled in TEST env, we have to search on user_name. Once that's enabled, we can add the test back